js/modulestest/runtime.go

```diff
@@ -74,6 +74,34 @@ func (r *Runtime) SetupModuleSystemFromAnother(another *Runtime) error {
 	return r.innerSetupModuleSystem()
 }
 
+// RunOnEventLoop will run the given code on the event loop.
+//
+// It is meant as a helper to test code that is expected to be run on the event loop, such
+// as code that returns a promise.
+//
+// A typical usage is to facilitate writing tests for asynchrounous code:
+//
+//	func TestSomething(t *testing.T) {
+//	    runtime := modulestest.NewRuntime(t)
+//
+//	    err := runtime.RunOnEventLoop(`
+//	        doSomethingAsync().then(() => {
+//	            // do some assertions
+//	        });
+//	    `)
+//	    require.NoError(t, err)
+//	}
+func (r *Runtime) RunOnEventLoop(code string) error {
```

Comment:
It seems like we already have something similar:
https://github.com/grafana/k6/blob/master/js/modules/k6/k6_test.go#L432-L439

So if we want to introduce that exposed method, we should keep only one implementation and usage. What do you think?

Reply:
We have  a bunch of similar functions scattered across the codebase indeed 🤝 Another occurence I'm aware of is the http package which has one to test the `asyncRequest`.

If you're suggesting we should also address the know use case in this PR, I rather agree 👍🏻 I'll give it a shot, let me know if you find more similar cases that we should also address 🙇🏻

Reply:
I've added tests accordingly for all the code I could find in the codebase that did some `EventLoop.Start` in one way or another 👍🏻